spec.html

```diff
@@ -29107,11 +29108,11 @@ <h1>%StringIteratorPrototype%.next ( )</h1>
             1. If _position_ &ge; _len_, then
               1. Set _O_.[[IteratedString]] to *undefined*.
               1. Return CreateIterResultObject(*undefined*, *true*).
-            1. Let _first_ be the numeric value of the code unit at index _position_ within _s_.
-            1. If _first_ &lt; 0xD800 or _first_ &gt; 0xDBFF or _position_ + 1 = _len_, let _resultString_ be the String value consisting of the single code unit _first_.
+            1. Let _first_ be the code unit at index _position_ within _s_.
+            1. If _first_ is not a <emu-xref href="#leading-surrogate"></emu-xref> or if _position_ + 1 = _len_, let _resultString_ be the String value consisting of the single code unit _first_.
             1. Else,
-              1. Let _second_ be the numeric value of the code unit at index _position_ + 1 within the String _s_.
-              1. If _second_ &lt; 0xDC00 or _second_ &gt; 0xDFFF, let _resultString_ be the String value consisting of the single code unit _first_.
+              1. Let _second_ be the code unit at index _position_ + 1 within the String _s_.
+              1. If _second_ is not a <emu-xref href="#trailing-surrogate"></emu-xref>, let _resultString_ be the String value consisting of the single code unit _first_.
               1. Else, let _resultString_ be the string-concatenation of the code unit _first_ and the code unit _second_.
```

Comment:
it kind of seems like we could benefit from an abstract operation that takes `s`, and `position`, and returns either the concatenation, or a List of `first, second` - then we could use it both here, and in codePointAt (and potentially other places)

Reply:
It’d be nice to get that in this PR, if possible ;-)

Reply:
Added in 9105de57f9ab1ada4337723125d52db4bc1ce385, though I'm not sure whether the new operation belongs with UTF16Encoding and UTF16Decode under [Section 10.1: Source Text](https://tc39.github.io/ecma262/#sec-source-text) or somewhere else, or even whether all three should move to something like a new "Code Point Operations" section under [Section 7: Abstract Operations](https://tc39.github.io/ecma262/#sec-abstract-operations). Please share your thoughts.

Reply:
Thanks, this is great. i don’t have an opinion on where it lives; alongside the utf16 ops seems fine for now.